### ISSUE TITLE
Improving Sequence-to-Sequence Learning via Optimal Transport

### ISSUE THREAD

**Issue by ymgw55** [ymgw55/MLPapers#40]
# Information
### URL
https://arxiv.org/abs/1901.06283
### Authors
Liqun Chen, Yizhe Zhang, Ruiyi Zhang, Chenyang Tao, Zhe Gan, Haichao Zhang, Bai Li, Dinghan Shen, Changyou Chen, Lawrence Carin
### Published at
Microsoft
### Date of issue
18 Jan 2019
### Code
Tensorflow (official) : https://github.com/LiqunChen0606/OT-Seq2Seq
PyTorch: https://github.com/ymgw55/OT-Seq2Seq-PyTorch
# Reading Motivation
#6 でpolicy gradientは分散が大きいのでその回避手法としてOT loss のback propagationを提案した論文として紹介されていた.
# Summary
MLE は次の単語を予測することにfocusしているが,
言語モデルの損失関数にOT項を足すことで, 文章同士の比較ができる.

# Contribution
soft-argmax を用いることで確率最大のtokenを選ぶという操作が連続になり, OT loss を更新する際に微分できる.
![Screenshot from 2021-05-11 22-03-50](https://user-images.githubusercontent.com/39965001/117819758-d1d96c80-b2a4-11eb-8c2d-1b2e82385cb3.png)
